src/hevm/src/EVM.hs

```diff
@@ -675,13 +675,15 @@ exec1 = do
                                      -- Instead, we keep track of all of the particular invocations of symkeccak' we see
                                      -- (similarly to sha3Crack), and simply assert that injectivity holds for these
                                      -- particular invocations.
+                                     --
+                                     -- TODO - maybe here Dull needs to be replaced with something smarrrtttt
 
                                      SymbolicBuffer bs -> do
                                                    let hash' = symkeccak' bs
                                                        previousUsed = view (env . keccakUsed) vm
                                                    env . keccakUsed <>= [(bs, hash')]
-                                                   pathConditions <>= fmap (\(preimage, image) ->
-                                                                               image .== hash' .=> preimage .== bs)
+                                                   constraints <>= fmap (\(preimage, image) ->
+                                                                               (image .== hash' .=> preimage .== bs, Dull))
```

Comment:
can't we use `FromKeccak preimage` instead of `Dull` here?

Reply:
kind of.... It's more like a injectivity constraint on keccak than a fromKeccak